### ISSUE TITLE
Calendar can pick up the wrong day, depending on the time

### ISSUE THREAD

**Issue by stevejburge** [publishpress/PublishPress#572]
Reported by a user and confirmed in my testing ...

s creating posts with a Scheduled status allowed from the calendar? I created this post (at 7 PM) and scheduled it for 11:30 PM today, see the 2 screenshots below.

![image](https://user-images.githubusercontent.com/3868028/74558523-70571080-4f30-11ea-9dab-54f553555f71.png)

The resulting post from above says its's already published, the exception this seems to be a post scheduled 24 hours later or so...

![image-1](https://user-images.githubusercontent.com/3868028/74558540-7816b500-4f30-11ea-93aa-7b75dc8edac7.png)

**Comment by andergmartins** [comment]
This is probably related to timezone.

**Comment by stevejburge** [comment]
Yes, looks like a 5 hour mismatch. I think this was UK site and a server on the East Coast of the US.

**Comment by andergmartins** [comment]
Humm, so if we confirm this is a misconfiguration issue maybe we could try to implement a warning for the admins?

This kind of problem already consumed a lot of time in other issues in the past. Maybe showing a warning in case we detect a difference between the timezone settings? We can try to compare the server's timezone with WordPress' time zone.

But I need to isolate the issue and check if this is something that happens in WordPress normally, or only when PP is activated.

**Comment by stevejburge** [comment]
@andergmartins Yes, that's a good idea. We could show it in the Settings, perhaps in a Debug area.

**Comment by andergmartins** [comment]
I just replicated the issue. It seems like it is not related to the server's timezone, but only to the timezone set in WP. 
I was testing using "America/Sao_Paulo" (GMT-03:00).

I scheduled a new post in the calendar without set the time, so it is added as "00:00". The post is displayed in the correct day, but after refreshing the page it displays on the day before, with the scheduled time as "21:00".

When saving in the calendar it seems like to be considering the "00:00" time as "GMT" time, and applying the WP timezone in the time value, modifying it. The time shouldn't be updated but stored as the user added it.
